coderd/httpmw/oauth2.go

```diff
@@ -114,6 +114,15 @@ func ExtractOAuth2(config OAuth2Config) func(http.Handler) http.Handler {
 				redirect = stateRedirect.Value
 			}
 
+			ctx, err = codersdk.HandleOauth2ClientCertificates(ctx, tlsConfig)
```

Comment:
This has the unfortunate side-effect of creating a new http client for each oauth request. It would be better to pass the client/context some other way so we can always use the same instance.

Reply:
I 100% agree

I'm relatively new to Go

Is there somewhere where the server context is initialized once and I can add this to the context there?

Or is it more like a, I create the http client statically once in `coderd.go` and then add it to the context of every request that comes in? (or at least every request in `/oauth/*`)